simulator/source/Combat_simulator.cpp

```diff
@@ -1510,6 +1564,20 @@ void Combat_simulator::normal_phase(Sim_state& state, bool mh_swing)
             return;
         }
     }
+        
+    if (use_devastate_)
+    {
+        bool dt_ww = true;
+        if (config.combat.use_whirlwind)
+        {
+            dt_ww = std::max(time_keeper_.whirlwind_cd(), 100) > config.combat.devastate_whirlwind_cooldown_thresh;
```

Comment:
If `devastate_whirlwind_cooldown_thresh` is at its default of 0, `dt_ww` is always set here, and Devastate should push WW off the table (it always costs less than WW).
I'd simply move the "devastate block" below WW, so WW is always preferred if ready (it should always be more damage per global), and devastate /might/ yield to it if WW's CD is ending soon.

Reply:
Yeah, I noticed the behavior where WW was pushed off the table by devastate with the default cooldown threshold value in my testing. I realized that this is just an inherent feature of the way the sim works, if the cooldown thresholds of both whirlwind and BT/MS/Devastate are set to zero then whenever both cooldowns are up the order is just determined by the the order of these statements in the code. The only way that a user can force priority of one over the other in that case is to enter in a small number one skill's threshold so that the other skill takes priority. This is obviously much more important with devastate since it has no cooldown. It didn't necessarily seem like an improper behavior to me, but I agree that your solution is probably necessary since the user checking the "use WW" box implies that they actually want to cast it and we should probably make sure that happens by default without any other inputs being necessary.